modules/navier_stokes/src/auxkernels/NSMachAux.C

```diff
@@ -0,0 +1,50 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+
+#include "NSMachAux.h"
+#include "MooseMesh.h"
+
+template<>
+InputParameters validParams<NSMachAux>()
+{
+  InputParameters params = validParams<AuxKernel>();
+
+  // Mark variables as required
+  params.addRequiredCoupledVar("u", "x-velocity");
+  params.addRequiredCoupledVar("v", "y-velocity");
+  params.addCoupledVar("w", "z-velocity"); // Only required in 3D...
+  params.addRequiredCoupledVar("temperature", "");
+
+  // Parameters with default values
+  params.addRequiredParam<Real>("gamma", "Ratio of specific heats");
+  params.addRequiredParam<Real>("R", "Gas constant.");
+
+  return params;
+}
+
+NSMachAux::NSMachAux(const InputParameters & parameters) :
+    AuxKernel(parameters),
+    _u_vel(coupledValue("u")),
+    _v_vel(coupledValue("v")),
+    _w_vel(_mesh.dimension() == 3 ? coupledValue("w") : _zero),
```

Comment:
> if you go with isCoupled() and somebody forgets to couple something in 3D, then you are going to get silently wrong results.

I know, unfortunately we do not have a better way how to guard against it ;-( we would need a map from block ID -> block dimension in something like `FEProblem` and then use that info in integrity checks. Or something along those lines...